test/fuzz/random.h

```diff
@@ -0,0 +1,79 @@
+#include <random>
+#include <vector>
+
+namespace Envoy {
+namespace Random {
+
+class PsuedoRandomGenerator64 : public RandomGenerator {
+public:
+  PsuedoRandomGenerator64() = default;
+  ~PsuedoRandomGenerator64() override = default;
+
+  void initializeSeed(uint64_t seed) { prng_ = std::make_unique<std::mt19937_64>(seed); }
+
+  // RandomGenerator
+  uint64_t random() override {
+    uint64_t to_return = (*prng_.get())();
+    ENVOY_LOG_MISC(trace, "random() returned: {}", to_return);
+    return to_return;
+  }
+  std::string uuid() override { return ""; }
+  std::unique_ptr<std::mt19937_64> prng_;
+};
+
+} // namespace Random
+
+namespace Fuzz {
+class ProperSubsetSelector {
+public:
+  ProperSubsetSelector(std::string random_bytestring) {
+    random_bytestring_ = random_bytestring;
+    index_of_random_bytestring_ = 0;
+  }
+
+  // Returns a vector of vectors, each vector of bytes representing a single subset
+  std::vector<std::vector<uint8_t>>
+  constructSubsets(uint32_t number_of_subsets,
+                   std::vector<uint32_t> number_of_elements_in_each_subset,
+                   uint32_t number_of_elements) {
+    ASSERT(number_of_subsets == number_of_elements_in_each_subset.size());
```

Comment:
highly recommend adding an example to the doc string or above the class.

Reply:
Haha, guess what, I had a ASSERT on the sum of all elements being < element in subset. However, this is handled in the for loop (uint32_t i = 0; i < number_of_elements_in_subset && index_vector_.size() != 0; i++) on the index_vector_.size() being zero. So thus, I removed the assert, since I like it like this. Number of elements is needed because that's what is used for the index vector! Thus, when it chooses an index to use "randomly" (based on bytestring), it needs the full index list, which is determined by element in each subset. A subset doesn't require every element!

Reply:
Added doc comment with an example and explanation.